cheri-c-programming.tex

```diff
@@ -1124,7 +1124,7 @@ \subsubsection{Restrictions in capability locations in memory}
   While the compiler and native allocators (stack, heap, \ldots{}) will
   provide sufficient alignment for capability-based pointers, custom
   allocators may align allocations to \ccode{sizeof(intmax\_t)} rather than
-  \ccode{sizeof(intptr\_t)}.
+  \ccode{sizeof(max\_align\_t)}.
```

Comment:
This isn't what you want.  `max_align_t` is `__max_align_t` which is:
```
typedef struct {
        long long __max_align1 __aligned(_Alignof(long long));
#ifndef _STANDALONE
        long double __max_align2 __aligned(_Alignof(long double));
#endif
        void * __max_align3 __aligned(_Alignof(void *));
} __max_align_t;
```
I still don't understand why gcc defined it as a struct, but they did... IIRC @bsdjhb might remember more.

Reply:
I fixed this to be alignof locally but haven't pushed that change yet.

Reply:
I had almost changed this sentence as well.  I'm also not sure why it's a struct instead of a union, but alignof should fix that.

Reply:
That is a rather subtle bug, maybe we should add a footnote stating that sizeof() is wrong?

Reply:
I think a footnote for that is a good idea.